Use one upload to script

roxell opened the pull request:
Use one upload script for both artifactorial and for squad.

roxell committed:
automated: utils: support squad in upload-to-artifactoral

Add support for upload to squad in upload-to-artifactoral.sh and create
a symlink upload-to-squad.sh to upload-to-artifactorial.sh.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

roxell committed:
automated: android: boottime: drop 'TOKEN'

Explaining that the artifactoral token should be set by the submitters
LAVA profile.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

roxell committed:
automated: android: apk-automation: drop '*_TOKEN'

Explaining that the artifactoral token should be set by the submitters
LAVA profile.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

roxell committed:
automated: utils: add script symlink upload-artifacts.sh

Add a more generic script name for uploading artifacts to artifactorial
or SQUAD.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

roxell committed:
automated: android: apk-automation: use upload-artifacts.sh

Use script upload-artifacts.sh, just have to specify the correct
UPLOAD_URL and set the correct TOKEN in the submitters profile.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

roxell committed:
automated: android: boottime: use upload-artifacts.sh

Use script upload-artifacts.sh, just have to specify the correct
UPLOAD_URL and set the correct TOKEN in the submitters profile.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

roxell committed:
automated: utils: remove script upload-to-squad

No need to keep symlink upload-to-squad.sh, upload-artifacts.sh should
be used instead.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

liuyq commented on automated/android/apk-automation/apk-automation.yaml:
```diff
@@ -27,16 +27,13 @@ params:
-    # SQUAD_ARCHIVE_SUBMIT_TOKEN is used for uploading authentication,
-    # and must be defined by the submitter as one profile managed token
-    SQUAD_UPLOAD_URL: ""
+    UPLOAD_URL: "https://archive.validation.linaro.org/artifacts/team/qa/"
```
This is related to the interface change, it would be better to support all of them for a while and drop them after that period.
Like the current status, there are jobs using ARTIFACTORIAL_URL(old jobs not working at the moment) and SQUAD_UPLOAD_URL(work now), and then will submit jobs to support "UPLOAD_URL" if we have this PR merged right now, then the jobs with SQUAD_UPLOAD_URL will fail, and it needs to resubmit them again. if both SQUAD_UPLOAD_URL and UPLOAD_URL are supported here, then the current jobs will be able to run, and it's possible to submit new jobs with UPLOAD_URL as well. ARTIFACTORIAL_URL support will be the similar.

liuyq commented on automated/android/apk-automation/apk-automation.yaml:
```diff
@@ -45,7 +42,6 @@ run:
         - tar caf "output-${TEST_NAME}.tar.xz" "./output"
-        - if [ -n "${SQUAD_UPLOAD_URL}" ]; then ../../utils/upload-to-squad.sh -a "output-${TEST_NAME}.tar.xz" -u "${SQUAD_UPLOAD_URL}"; fi
-        - if [ -z "${SQUAD_UPLOAD_URL}" ]; then ../../utils/upload-to-artifactorial.sh -a "output-${TEST_NAME}.tar.xz" -u "${ARTIFACTORIAL_URL}" -t "${ARTIFACTORIAL_TOKEN}"; fi
+        - ../../utils/upload-artifacts.sh -a "output-${TEST_NAME}.tar.xz" -u "${UPLOAD_URL}"
```
to support my concern, there could be done like this:
```
- if [ -n "${UPLOAD_URL}" ]; ../../utils/upload-artifacts.sh -a "output-${TEST_NAME}.tar.xz" -u "${UPLOAD_URL}"; fi
- if [ -n "${SQUAD_UPLOAD_URL}" ]; ../../utils/upload-artifacts.sh -a "output-${TEST_NAME}.tar.xz" -u "${SQUAD_UPLOAD_URL}"; fi
```
And after like several weeks or a month, drop the SQUAD_UPLOAD_URL line when no SQUAD_UPLOAD_URL jobs will be run.

roxell replied:
make sense, to keep it around for a while. Good catch.

mwasilew commented:
LGTM. I share @liuyq worries about backward compatibility. As I'm not using any uploads right now I'll leave this to you to decide.

roxell committed:
automated: android: noninteractive-tradefed: drop 'TOKEN'

Explaining that the artifactoral token should be set by the submitters
LAVA profile.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

roxell committed:
automated: android: noninteractive-tradefed: use upload-artifacts.sh

Use script upload-artifacts.sh, just have to specify the correct
UPLOAD_URL and set the correct TOKEN in the submitters profile.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

liuyq commented on automated/android/noninteractive-tradefed/tradefed.yaml:
```diff
@@ -70,8 +70,8 @@ run:
         - ATTACHMENT=$(ls tradefed-output-*.tar.xz)
-        - if [ -n "${SQUAD_UPLOAD_URL}" ]; then ../../utils/upload-to-squad.sh -a "${ATTACHMENT}" -u "${SQUAD_UPLOAD_URL}"; fi
-        - if [ -z "${SQUAD_UPLOAD_URL}" ]; then ../../utils/upload-to-artifactorial.sh -a "${ATTACHMENT}" -u "${URL}" -t "${TOKEN}"; fi
+        - if [ -n "${SQUAD_UPLOAD_URL}" ]; then UPLOAD_URL="${SQUAD_UPLOAD_URL}" fi
```
I am not quite sure, but I think it's this line that has the syntax error, like reported by the lava job here:
https://lkft.validation.linaro.org/scheduler/job/7371910#L8593

the right one should be like this:
`if [ -n "${SQUAD_UPLOAD_URL}" ]; then UPLOAD_URL="${SQUAD_UPLOAD_URL}"; fi`
